internal/method/resolver_test.go

```diff
@@ -220,6 +226,22 @@ func (mg *Model) ResolveReferences(ctx context.Context, c client.Reader) error {
 	mg.Spec.ForProvider.RouteTableIDs = reference.ToPtrValues(mrsp.ResolvedValues)
 	mg.Spec.ForProvider.RouteTableIDsRefs = mrsp.ResolvedReferences
 
+	rsp, err = r.Resolve(ctx, reference.ResolutionRequest{
```

Comment:
Could you please consider adding tests for the `generate.WriteMethods` function?

Reply:
In `WriteMethods` there is a file writing operation: 

`return errors.Wrap(ioutil.WriteFile(file, b.Bytes(), 0644), "cannot write Go file")`

It seems that when we want to write unit-tests to this function we will have to find a good way to handle this operation (such as mocking this operation or writing test files with temporary way).

It looks like you need to make changes to the production code to use a mock method. Because currently, the function accepts a filename and directly calls ioutil's Write function.

I suggest that this unit-test be followed by opening another issue together with the refactoring in the prod code. What are your thoughts @ulucinar